app/controllers/maintenance_tasks/runs_controller.rb

```diff
@@ -14,6 +14,7 @@ def create(&block)
         name: params.fetch(:task_id),
         csv_file: params[:csv_file],
         arguments: params.fetch(:task_arguments, {}).permit!.to_h,
+        metadata: MaintenanceTasks.metadata&.call,
```

Comment:
The README states that the `metadata` proc will be exec'd in the scope of the controller, but this isn't doing that -- did you intend to `instance_exec(MaintenanceTasks.metadata) if MaintenanceTasks.metadata` ?